zcash_client_sqlite/src/scan.rs

```diff
@@ -57,6 +57,38 @@ struct WitnessRow {
 pub fn scan_cached_blocks<P: AsRef<Path>, Q: AsRef<Path>>(
     db_cache: P,
     db_data: Q,
+) -> Result<(), Error> {
+    scan_cached_block_batch(db_cache, db_data, std::i32::MAX)
+}
+
+/// Scans new blocks added to the cache for any transactions received by the tracked
+/// accounts but does so in batches of limited size.
+///
+/// This function pays attention only to cached blocks with heights greater than the
+/// highest scanned block in `db_data`. Cached blocks with lower heights are not verified
+/// against previously-scanned blocks. In particular, this function **assumes** that the
+/// caller is handling rollbacks.
+///
+/// For brand-new light client databases, this function starts scanning from the Sapling
+/// activation height. This height can be fast-forwarded to a more recent block by calling
+/// [`init_blocks_table`] before this function.
+///
+/// Scanned blocks are required to be height-sequential. If a block is missing from the
+/// cache, an error will be returned with kind [`ErrorKind::InvalidHeight`].
+///
+/// # Examples
+///
+/// ```
+/// use zcash_client_sqlite::scan::scan_cached_blocks;
+///
+/// scan_cached_blocks("/path/to/cache.db", "/path/to/data.db");
+/// ```
+///
+/// [`init_blocks_table`]: crate::init::init_blocks_table
+pub fn scan_cached_block_batch<P: AsRef<Path>, Q: AsRef<Path>>(
+    db_cache: P,
+    db_data: Q,
+    limit: i32,
```

Comment:
Make this a `u32` (`LIMIT` requires non-negative integers).

Reply:
Could not get u32 to work with `query_map` I suspect this has to do with the underlying SQLite implementation only having signed integer types?